src/main/kotlin/com/jetbrains/snakecharm/inspections/SmkRuleRedeclarationInspection.kt

```diff
@@ -27,8 +29,14 @@ class SmkRuleRedeclarationInspection : SnakemakeInspection() {
         private fun visitSMKRuleLike(rule: SmkRuleLike<SmkArgsSection>) {
             val ruleName = rule.name ?: return
             if (ruleNames.contains(ruleName)) {
-                registerProblem(rule.nameIdentifier,
-                        SnakemakeBundle.message("INSP.NAME.rule.redeclaration"))
+                val problemElement = rule.nameIdentifier ?: return
+                registerProblem(
+                        problemElement,
+                        SnakemakeBundle.message("INSP.NAME.rule.redeclaration"),
+                        ProblemHighlightType.ERROR,
+                        null,
+                        PyRenameElementQuickFix(problemElement)
```

Comment:
Works weird and produce code which fails on runtime in case of 
```
ruleorder: boo > foo1
localrules: boo

rule boo:
    output: "aa"

rule foo1:
    input: rules.boo

rule boo:
    input: "aa"

rule foo2:
    input: rules.boo
```
Are you sure that it is expected behaviour?
Result is:
```
ruleorder: boo > foo1
localrules: boo

rule boo:
    output: "aa"

rule foo1:
    input: rules.aaaaaaaaaaa

rule aaaaaaaaaaa:
    input: "aa"

rule foo2:
    input: rules.aaaaaaaaaaa
```

Reply:
>Works weird and produce code which fails on runtime in case of
Please clarify what you mean by 'fails at runtime'. I thought  you might mean that there were exceptions but I tried to reproduce this and did not get any, so I'd like to ask for further clarification. 

I assume by 'weird' you mean the fact that the rule got renamed in `rules.` calls but not in localrules and ruleorder sections. I think this might be due to how resolve is implemented for both localrules/ruleorder and `rules.`: if we check usages of the first rule `boo`, we get both localrules/ruleorder and `rules.` usages; for the second rule `boo` we only get `rules`. I suppose this is an issue with resolve, not with this particular action.

It is possible to modify `SmkReferenceExpressionImpl` in a way that renaming of the second rule would also rename its usages in localrules/ruleorder section. However, I assume the desired behavior in this case (that is, for the second rule) is no renaming for usages at all? I am not sure how to implement that unless we abandon multiresolve for `AbstractSmkRuleOrCheckpointType` and `SmkReferenceExpressionImp`, because otherwise even a simple rename action invocation (not in regards to this inspection, just plain old Shift + F6) on the second rule leads to renaming of all its usages. Modifying `SmkReferenceExpressionImpl` is still worth it though, I think, because as of now we probably have an issue with detecting usages?

Anyway, I don't think it's related to this quick fix, but rather to resolve? Unless you were thinking of a completely different problem and I misunderstood your comment.

Reply:
done